src/lib/define-dynamic-block.js

```diff
@@ -2,6 +2,62 @@
 // Should we move these into a new extension support module or something?
 import ArgumentType from 'scratch-vm/src/extension-support/argument-type';
 import BlockType from 'scratch-vm/src/extension-support/block-type';
+import ContextMenuContext from 'scratch-vm/src/extension-support/context-menu-context';
+import log from './log.js';
+
+const setupCustomContextMenu = (ScratchBlocks, contextMenuInfo, extendedOpcode) => {
+    // Handle custom context menu options
+    const customContextMenuForBlock = {
+        customContextMenu: function (options) {
+            contextMenuInfo.forEach(contextOption => {
+                const option = {
+                    enabled: true,
+                    text: contextOption.text,
+                    callback: () => {
+                        if (contextOption.builtInCallback) {
+                            switch (contextOption.builtInCallback) {
+                            case 'EDIT_A_PROCEDURE':
+                                // TODO FILL THIS IN
```

Comment:
What kind of thing do you imagine going here? Firing an event maybe? Or will this be a direct call into `ScratchBlocks`?

Reply:
In this block of code (and the associated `RENAME_A_VARIABLE` block) you use the gui context to open the appropriate modal (e.g. edit modal or rename modal) and then handle the data submitted from the modal and eventually make a call to the contextOption.callback so that the extension callback also has access to the data submitted from the modal.